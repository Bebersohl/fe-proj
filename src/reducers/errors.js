const errors = (state = {}, action) => {
  const { type, error } = action
  if (type === 'RESET_ERRORS') {
    return {}
  } else if (
    error &&
    error.hasOwnProperty('code') &&
    error.hasOwnProperty('message')
  ) {
    return {
      ...state,
      [error.code]: error.message,
    }
  }

  return state
}

export default errors
