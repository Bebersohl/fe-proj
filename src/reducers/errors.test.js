import errors from './errors'

it('handles initial state', () => {
  const next = errors(undefined, {})
  expect(next).toMatchSnapshot()
})

it('handles adding errors', () => {
  const next = errors({}, {type: 'TEST', error: {code: 'errorCode', message: 'error message'}})
  expect(next).toMatchSnapshot()
})

it('handles RESET_ERRORS', () => {
  const next = errors({error: 'error'}, {type: 'RESET_ERRORS'})
  expect(next).toMatchSnapshot()
})
