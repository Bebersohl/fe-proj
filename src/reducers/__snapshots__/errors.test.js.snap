exports[`test handles RESET_ERRORS 1`] = `Object {}`;

exports[`test handles adding errors 1`] = `
Object {
  "errorCode": "error message",
}
`;

exports[`test handles initial state 1`] = `Object {}`;
